tetris/src/main/java/com/sparklicorn/bucket/tetris/util/structs/ShapeQueue.java

```diff
@@ -127,8 +127,10 @@ public String toString() {
 
 	/** Throws UnsupportedOperationException */
 	@Override public boolean contains(Object o) { throw new UnsupportedOperationException(); }
-	/** Throws UnsupportedOperationException */
-	@Override public Iterator<Shape> iterator() { throw new UnsupportedOperationException(); }
+
+	@Override public Iterator<Shape> iterator() {
+		return shapeIndexQueue.stream().map((shapeIndex) -> Shape.getShape(shapeIndex)).iterator();
+	}
```

Comment:
Json serialization may be used in testing tools, which requires a functioning iterator.